modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/home/components/ProjectsList/ProjectList.test.js

```diff
@@ -0,0 +1,114 @@
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import {faker} from '@faker-js/faker';
+import {render, screen} from '@testing-library/react';
+import ProjectList from '.';
+
+describe('render Project List component', () => {
+	const koroneikiAccounts = {
+		items: [
+			{
+				accountKey: `KOR-${faker.random.numeric(5)}`,
+				code: faker.commerce.product().toUpperCase(),
+				name: faker.commerce.productName(),
+				region: faker.address.country(),
+				slaCurrentEndDate: faker.date.future(),
+				status: 'Active',
+			},
+			{
+				accountKey: `KOR-${faker.random.numeric(5)}`,
+				code: faker.commerce.product().toUpperCase(),
+				name: faker.commerce.productName(),
+				region: faker.address.country(),
+				slaCurrentEndDate: faker.date.future(),
+				status: 'Active',
+			},
+			{
+				accountKey: `KOR-${faker.random.numeric(5)}`,
+				code: faker.commerce.product().toUpperCase(),
+				name: faker.commerce.productName(),
+				region: faker.address.country(),
+				slaCurrentEndDate: faker.date.future(),
+				status: 'Active',
+			},
+			{
+				accountKey: `KOR-${faker.random.numeric(5)}`,
+				code: faker.commerce.product().toUpperCase(),
+				name: faker.commerce.productName(),
+				region: faker.address.country(),
+				slaCurrentEndDate: faker.date.future(),
+				status: 'Active',
+			},
+			{
+				accountKey: `KOR-${faker.random.numeric(5)}`,
+				code: faker.commerce.product().toUpperCase(),
+				name: faker.commerce.productName(),
+				region: faker.address.country(),
+				slaCurrentEndDate: faker.date.future(),
+				status: 'Active',
+			},
+			{
+				accountKey: `KOR-${faker.random.numeric(5)}`,
+				code: faker.commerce.product().toUpperCase(),
+				name: faker.commerce.productName(),
+				region: faker.address.country(),
+				slaCurrentEndDate: faker.date.future(),
+				status: 'Active',
+			},
+		],
+		totalCount: 6,
+	};
+
+	window.IntersectionObserver = jest.fn(() => ({
+		observer: jest.fn(),
+		unobserver: jest.fn(),
+	}));
+
+	test('Verify if the message "results not found" appears if there is no project to show', () => {
+		render(<ProjectList />);
+
+		const showNotFoundMessage = screen.getByText(/no results found/i);
+		expect(showNotFoundMessage).toBeInTheDocument();
+	});
+
+	test('Verify if the projects are displayed as cards if has less than 05 projects', () => {
+		const compressed = false;
+
+		const {container} = render(
+			<ProjectList
+				hasManyProjects={compressed}
+				koroneikiAccounts={koroneikiAccounts}
+				totalCount={koroneikiAccounts.totalCount}
+			/>
+		);
+
+		expect(
+			container.getElementsByClassName('cp-home-projects').length
```

Comment:
We should get an element by its classname only if that is the only way to do that, but it can be tricky sometimes
This is an "implementation detail" because we restrict the test to a development thing
Imagine if we only change the class' name but the behavior of the component remains the same, the test will fail

Reply:
Hey Aildson, thanks for the observation. Actually this the only way I found to get this element because we just changed the classname when compressed is true, and then we change the styles, but still have the same div. Do you  suggest a better way to do this?